activerecord/lib/active_record/reflection.rb

```diff
@@ -741,7 +756,12 @@ def association_class
       # klass option is necessary to support loading polymorphic associations
       def association_primary_key(klass = nil)
         if primary_key = options[:primary_key]
-          @association_primary_key ||= -primary_key.to_s
+          @association_primary_key ||= if primary_key.is_a?(Array)
+            # composite foreign keys support
+            Array.wrap(primary_key).freeze
```

Comment:
It doesn't cause an error but on line 763 we `to_s` the primary keys, but here we don't. In `last_scope_chain` when these are read, the primary keys are symbols and the foreign keys are strings. In a standard, non-composite belongs_to, they are both strings.

Reply:
Good catch, it is going to cause an error in a different part of the functionality (association assignments)  I'm working on in a separate branch and I have it fixed there. I'm going to cherry-pick the change into this PR, thanks! 👍